libraries/AP_UAVCAN/AP_UAVCAN.cpp

```diff
@@ -1347,4 +1347,16 @@ bool AP_UAVCAN::led_write(uint8_t led_index, uint8_t red, uint8_t green, uint8_t
     return true;
 }
 
+bool AP_UAVCAN::is_driver_ready(uint8_t mgr)
+{
+    if (hal.can_mgr[mgr] == nullptr) {
+    	return false;
+    }
+    AP_UAVCAN *ap_uavcan = hal.can_mgr[mgr]->get_UAVCAN();
+    if (ap_uavcan == nullptr) {
+        return false;
+    }
+    return true;
```

Comment:
I was actually talking about:

```c++
// Return uavcan from @iface or `nullptr` if it's not ready or doesn't exist
AP_UAVCAN * AP_UAVCAN::get_uavcan(uint8_t iface)
{
    if (iface >= MAX_NUMBER_OF_CAN_INTERFACES || !hal.can_mgr[iface]) {
        return nullptr;
    }
    return hal.can_mgr[iface]->get_UAVCAN();
}
```

Reply:
ahh, so move it all into here and then still do a single null check in the calling function?

Reply:
There should be a check in calling func anyway. So we can combine assign with if in one line.

Reply:
@lucasdemarchi  I'm having a bad day and think we're all better off if I walk away from the computer for a while. Feel free to force-push to this branch with whatever idea you have

Reply:
"Add more commits by pushing to the CAN/check_nulls branch on magicrub/ardupilot."

pushing to **your** branch? I didn't know I can do that

Reply:
That is possible now :)